File: 0238-product-of-array-except-self/0238-product-of-array-except-self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
       int prod=1;
       int count_zero=0;
       for(int i:nums){
        if(i!=0)prod*=i;
        else count_zero++;
        }

       int ar[]= new int[nums.length];
       for(int i=0;i<nums.length;i++){
        if(count_zero>1)
        ar[i]=0;
        else if(nums[i]==0)
        ar[i]=prod;
        else if(count_zero==0)
        ar[i]=prod/nums[i];
        else
        ar[i]=0;
       }

       return ar; 
    }
}

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var productExceptSelf = function(nums) {
    let zero = 0;
    let sum = 1;

    for(const it of nums){ 
        if(it !=0)sum *= it;
        else zero++;
    }

    let arr = [];

    for(const it of nums){
        if(zero >= 2)arr.push(0);
        else if(zero == 1 && it == 0)arr.push(sum);
        else if(zero == 1)arr.push(0);
        else arr.push(sum/it);
    }
    return arr;

};